### PR TITLE
cli: update frontend plugin template to use new extension API

### PR DESCRIPTION
**Pull request by Rugvip**
This switches the `create-plugin` template to use the new composability API and naming schemes from https://backstage.io/docs/plugins/composability.

### CODE REVIEW

File: packages/cli/templates/default-plugin/src/plugin.ts.hbs [hunk 1]
```diff
@@ -1,14 +1,18 @@
...
+import { createPlugin, createRoutableExtension } from '@backstage/core';
 
-export const rootRouteRef = createRouteRef({
-  path: '/{{ id }}',
-  title: '{{ id }}',
-});
+import { rootRouteRef } from './routes';
 
-export const plugin = createPlugin({
+export const {{ pluginVar }} = createPlugin({
```
Comment (freben) [inline packages/cli/templates/default-plugin/src/plugin.ts.hbs:5]: This should mean updating docs, right? For plugin.ts

**Review by freben — APPROVED** [review]
So much nicer

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4251-sr6v40rioipwjzvgqclivcmhgk1grqn0.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60134a24353bbb80b59a8702

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4251-sr6v40rioipwjzvgqclivcmhgk1grqn0.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60134a24353bbb80b59a8702

File: packages/cli/src/commands/create-plugin/createPlugin.ts [hunk 2]
```diff
@@ -126,6 +124,46 @@ export async function addPluginToApp(
   });
 }
 
+export async function addPluginExtensionToApp(
+  pluginId: string,
+  extensionName: string,
+  pluginPackage: string,
+) {
+  const pluginsFilePath = paths.resolveTargetRoot('packages/app/src/app.tsx');
```
Comment (freben) [inline packages/cli/src/commands/create-plugin/createPlugin.ts:132]: Case insensitive?

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4251-sr6v40rioipwjzvgqclivcmhgk1grqn0.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60134a24353bbb80b59a8702

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4251-sr6v40rioipwjzvgqclivcmhgk1grqn0.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60134a24353bbb80b59a8702

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4251-sr6v40rioipwjzvgqclivcmhgk1grqn0.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60134a24353bbb80b59a8702

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4251-sr6v40rioipwjzvgqclivcmhgk1grqn0.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60134a24353bbb80b59a8702

**Comment by freben** [comment]
Good to go?

**Pull request merged**